kratos/includes/condition.h

```diff
@@ -927,6 +927,30 @@ class Condition : public GeometricalObject, public Flags
 	  rDampingMatrix.resize(0, 0, false);
     }
 
+
+     /**
+     * Calculate the transposed gradient of the condition's residual w.r.t. design variable.
+     */
+    virtual void CalculateSensitivityMatrix(const Variable<double>& rDesignVariable,
```

Comment:
You need to discuss with @KratosMultiphysics/technical-committee  to add methods to the condition.h and element.h

Reply:
@loumalouomega thanks for pointing that out, I did now also request a review from the technical-committee. 
This function is the same as the one that was added to the element.h according to the wiki: https://github.com/KratosMultiphysics/Kratos/wiki/Adjoint-API

Reply:
I am ok with the technical change. But i would propose putting this in as a separate PR

(this one is way too big, so it is easy to have it unnoticed)